lib/new_relic/agent/llm/response_headers.rb

```diff
@@ -43,11 +52,28 @@ def populate_openai_response_headers(headers)
           self.llm_version = headers[OPENAI_VERSION]&.first
           self.rate_limit_requests = headers[X_RATELIMIT_LIMIT_REQUESTS]&.first
           self.rate_limit_tokens = headers[X_RATELIMIT_LIMIT_TOKENS]&.first
+          remaining_headers(headers)
+          reset_headers(headers)
+          tokens_usage_based_headers(headers)
```

Comment:
The branch condition for the method was getting to be too high, so I grouped headers with similar names into their own methods